### PR TITLE
Update TorchAgent with standard functions and fix small bug

### PR DESCRIPTION
**Pull request by vgonzalez88**
Fixes small bug where GPU wasn't specified and moves in the `save`, `load`, and `shutdown` functions with some very basic functionality.

### CODE REVIEW

File: parlai/core/torch_agent.py [hunk 1]
```diff
@@ -271,3 +275,34 @@ def parse(txt, splitSentences):
                 self.history['labels'] = labels
 
         return self.history['dialog']
+
+        def save(self, path):
```
Comment (jaseweston) [inline parlai/core/torch_agent.py]: save should save the opt file as well, see e.g. seq2seq

File: parlai/core/torch_agent.py [hunk 2]
```diff
@@ -75,6 +77,8 @@ def __init__(self, opt, shared=None):
         self.history_tokens = opt['history_tokens']
         self.history_dialog = opt['history_dialog']
         self.history_replies = opt['history_replies']
+        self.model = None
+        self.optimizer = None
```
Comment (alexholdenmiller) [inline parlai/core/torch_agent.py]: I don't think we should set these to None

File: parlai/core/torch_agent.py [hunk 3]
```diff
@@ -271,3 +275,34 @@ def parse(txt, splitSentences):
...
+            """Save model parameters if model_file is set.
+
+            Override this method for more specific saving.
+            """
+            path = self.opt.get('model_file', None) if path is None else path
+
+            if path and hasattr(self, 'model'):
+                model = {}
+                model['model'] = self.model.state_dict()
+                model['optimizer'] = self.optimizer.state_dict()
```
Comment (alexholdenmiller) [inline parlai/core/torch_agent.py]: this would need a `hasattr(self, 'optimizer')`

File: parlai/core/torch_agent.py (hunk 1 above)
Comment (alexholdenmiller) [reply parlai/core/torch_agent.py]: specifically, the opt should be saved in a separate file at modelfile.opt
let's also remove line 290 where we save it within the `model` dict

Comment (vgonzalez88) [reply parlai/core/torch_agent.py]: oh good point. do you think it's fine that `load` doesn't load in the opt file? I saw in seq2seq it does so, but didn't know if I should put that in the parent class

File: parlai/core/torch_agent.py (hunk 2 above)
Comment (vgonzalez88) [reply parlai/core/torch_agent.py]: Should I leave them out and check for both 'model' and 'optimizer' in the save function? Or is it safe to assume that the child class will have a 'model' field?

Comment (vgonzalez88) [reply parlai/core/torch_agent.py]: Oh I already was checking for 'model'. I'll just remove these.

**Comment by alexholdenmiller** [comment]
made a few changes myself :) looks good now!

**Comment by vgonzalez88** [comment]
Thanks Alex!

**Pull request merged**